src/components/flyout/flyout.styles.ts

```diff
@@ -0,0 +1,282 @@
+import { euiFormMaxWidth } from '../form/form.styles';
+
+const euiFlyout = keyframes`
+  0% {
+    opacity: 0;
+    transform: translateX(100%);
+  }
+  75% {
+    opacity: 1;
+    transform: translateX(0%);
+  }
+`;
+
+const euiFlyoutLeft = keyframes`
+  0% {
+    opacity: 0;
+    transform: translateX(-100%);
+  }
+  75% {
+    opacity: 1;
+    transform: translateX(0%);
+}
+`;
+
+export const euiFlyoutCloseButtonStyles = (euiThemeContext: UseEuiTheme) => {
+  const euiTheme = euiThemeContext.euiTheme;
+
+  return {
+    euiFlyout__closeButton: css`
+      background-color: ${transparentize(euiTheme.colors.emptyShade, 0.1)};
+      position: absolute;
+      ${logicalCSS('right', euiTheme.size.s)}
+      ${logicalCSS('top', euiTheme.size.s)}
+      z-index: 3;
+    `,
+    'closeButton--outside': css`
+      // match dropshadow
+      ${euiShadowXLarge(euiThemeContext)};
+      ${logicalCSS('right', 'auto')}
+      ${logicalCSS('left', 0)}
+      // Override the hover and focus transitions of buttons
+      animation: none !important;
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        transform: translateX(calc(-100% - ${euiTheme.size.l}));
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        transform: translateX(calc(-100% - ${euiTheme.size.xs}));
+      }
+    `,
+    'closeButton--inside': css``,
+    'closeButton--outside-left': css`
+      ${logicalCSS('right', 0)}
+      ${logicalCSS('left', 'auto')}
+      
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        transform: translateX(calc(100% + ${euiTheme.size.l}));
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        transform: translateX(calc(100% + ${euiTheme.size.xs}));
+      }
+    `,
+  };
+};
+
+export const euiFlyoutStyles = (
+  euiThemeContext: UseEuiTheme,
+  paddingSize: EuiFlyoutPaddingSize
+) => {
+  const euiTheme = euiThemeContext.euiTheme;
+
+  const euiFormMaxWidthNumber: number = parseInt(
+    euiFormMaxWidth(euiThemeContext).replace('px', ''),
+    10
+  );
+
+  const euiSizeMediumNumber: number = parseInt(
+    euiTheme.size.m.replace('px', ''),
+    10
+  );
+
+  const flyoutSizes = {
+    s: {
+      min: Math.round(euiTheme.breakpoint.m * 0.5),
+      width: '25vw',
+      max: Math.round(euiTheme.breakpoint.s * 0.7),
+    },
+
+    m: {
+      min: euiFormMaxWidthNumber + euiSizeMediumNumber * 2,
+      width: '50vw',
+      max: euiTheme.breakpoint.m,
+    },
+
+    l: {
+      min: Math.round(euiTheme.breakpoint.m * 0.9),
+      width: '75vw',
+      max: Math.round(euiTheme.breakpoint.l),
+    },
+  };
+
+  return {
+    euiFlyout: css`
+      ${logicalCSS('border-left', euiTheme.border.thin)}
+      ${euiShadowXLarge(euiThemeContext)};
+      position: fixed;
+      ${logicalCSS('top', 0)}
+      ${logicalCSS('right', 0)}
+      ${logicalCSS('bottom', 0)}
+      ${logicalCSS('height', '100%')}
+      z-index: ${euiTheme.levels.header};
```

Comment:
It looks like there's a z-index level specifically for flyouts, so we should probably use that
```suggestion
      z-index: ${euiTheme.levels.flyout};
```